api/python/quilt3/data_transfer.py

```diff
@@ -172,7 +172,8 @@ def register_signals(self, s3_client):
 
     def _build_client(self, get_config):
         session = self.get_boto_session()
-        return session.client('s3', config=get_config(session))
+        endpoint_url = os.environ.get('ENDPOINT_URL', "").strip() or None
```

Comment:
I wouldn't bother with `strip()` - I think it's good to be strict here - but otherwise, looks good.

```suggestion
        endpoint_url = os.environ.get('ENDPOINT_URL') or None
```

Reply:
@dimaryaz done :)